geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/StatSerializer.scala

```diff
@@ -273,15 +281,16 @@ object KryoStatSerializer {
     }
   }
 
-  private [stats] def readZ3Histogram(input: Input, sft: SimpleFeatureType, immutable: Boolean): Z3Histogram = {
+  private [stats] def readZ3Histogram(input: Input, sft: SimpleFeatureType, immutable: Boolean, version: Int): Z3Histogram = {
     val geomIndex = input.readInt(true)
     val dtgIndex  = input.readInt(true)
+    val period = if (version > 1) TimePeriod.withName(input.readString()) else TimePeriod.Week
     val length = input.readInt(true)
 
     val stat = if (immutable) {
-      new Z3Histogram(geomIndex, dtgIndex, length) with ImmutableStat
+      new Z3Histogram(geomIndex, dtgIndex, period, length) with ImmutableStat
     } else {
-      new Z3Histogram(geomIndex, dtgIndex, length)
+      new Z3Histogram(geomIndex, dtgIndex, period, length)
```

Comment:
there's not a time bound, but the Z3-related stats partition things by time period. Internally we have a stat object (like a count min sketch or whatever) for each e.g. week, which results in less error when estimating counts. When we store the stat, we store each time period separately. Previously it was always weeks, but now it's tied to the Z3 interval.